webapi/src/it/scala/org/knora/webapi/responders/admin/PermissionsResponderADMSpec.scala

```diff
@@ -681,341 +685,362 @@ class PermissionsResponderADMSpec extends CoreSpec with ImplicitSender with Priv
+          permissionIri = perm002_a1.iri,
+          requestingUser = rootUser
+        )
+        expectMsg(AdministrativePermissionGetResponseADM(perm002_a1.p))
+      }
+
+      "return a default object access permission" in {
+        appActor ! PermissionByIriGetRequestADM(
+          permissionIri = perm002_d1.iri,
+          requestingUser = rootUser
+        )
+        expectMsg(DefaultObjectAccessPermissionGetResponseADM(perm002_d1.p))
+      }
+    }
+
+    "ask to update group of a permission" should {
+      "update group of an administrative permission" in {
+        val permissionIri = "http://rdfh.ch/permissions/00FF/a2"
+        val newGroupIri   = "http://rdfh.ch/groups/00FF/images-reviewer"
+        appActor ! PermissionChangeGroupRequestADM(
+          permissionIri = permissionIri,
+          changePermissionGroupRequest = ChangePermissionGroupApiRequestADM(
+            forGroup = newGroupIri
+          ),
+          requestingUser = rootUser,
+          apiRequestID = UUID.randomUUID()
+        )
+        val received: AdministrativePermissionGetResponseADM = expectMsgType[AdministrativePermissionGetResponseADM]
+        val ap                                               = received.administrativePermission
+        assert(ap.iri == permissionIri)
+        assert(ap.forGroup == newGroupIri)
+      }
+
+      "throw ForbiddenException for PermissionChangeGroupRequestADM if requesting user is not system or project Admin" in {
+        val permissionIri = "http://rdfh.ch/permissions/00FF/a2"
+        val newGroupIri   = "http://rdfh.ch/groups/00FF/images-reviewer"
+        appActor ! PermissionChangeGroupRequestADM(
+          permissionIri = permissionIri,
+          changePermissionGroupRequest = ChangePermissionGroupApiRequestADM(
+            forGroup = newGroupIri
+          ),
+          requestingUser = SharedTestDataADM.imagesUser02,
+          apiRequestID = UUID.randomUUID()
+        )
+        expectMsg(
+          Failure(
+            ForbiddenException(
+              s"Permission $permissionIri can only be queried/updated/deleted by system or project admin."
+            )
+          )
+        )
+      }
+
+      "update group of a default object access permission" in {
+        val permissionIri = "http://rdfh.ch/permissions/00FF/d1"
+        val newGroupIri   = "http://rdfh.ch/groups/00FF/images-reviewer"
+        appActor ! PermissionChangeGroupRequestADM(
+          permissionIri = permissionIri,
+          changePermissionGroupRequest = ChangePermissionGroupApiRequestADM(
+            forGroup = newGroupIri
+          ),
+          requestingUser = rootUser,
+          apiRequestID = UUID.randomUUID()
+        )
+        val received: DefaultObjectAccessPermissionGetResponseADM =
+          expectMsgType[DefaultObjectAccessPermissionGetResponseADM]
+        val doap = received.defaultObjectAccessPermission
+        assert(doap.iri == permissionIri)
+        assert(doap.forGroup.get == newGroupIri)
```

Comment:
to get rid of the codacy warning, you could do this:

```suggestion
        assert(doap.forGroup == Some(newGroupIri))
```